src/plugins/intel_cpu/src/nodes/executors/dnnl/dnnl_utils.cpp

```diff
@@ -39,6 +40,39 @@ MemoryPtr prepareWeightsMemory(const DnnlMemoryDescPtr srcWeightDesc,
     }
 
     auto create = [&]() {
+        // https://oneapi-src.github.io/oneDNN/dev_guide_int8_computations.html?highlight=128#inputs-of-the-same-type-s8
+        auto src_wdt = srcWeightDesc->getPrecision();
+        auto dst_wdt = dstWeightDesc->getPrecision();
+        if (needShiftSignedToUnsigned && src_wdt.is_integral_number() && src_wdt.is_signed() &&
```

Comment:
FYI, Here is the tricky logic added to fallback to u8/u4 with zp 128/8:
 - we reorder layout of src into dst w/o type conversion
 - after layout is made right, we shift i8/i4 value by adding zero-point to make them into u8/u4 range, as documented in https://oneapi-src.github.io/oneDNN/dev_guide_int8_computations.html?highlight=128#inputs-of-the-same-type-s8